### PR TITLE
Add more specific share error

### PR DESCRIPTION
**Pull request by leikkisa**
connects https://github.com/department-of-veterans-affairs/caseflow/issues/12158

### CODE REVIEW

File: lib/bgs/errors.rb [hunk 1]
```diff
@@ -103,6 +108,19 @@ def initialize(message, code = nil)
     def ignorable?
       TRANSIENT_ERRORS.any? { |transient_error| message.include?(transient_error) }
     end
+
+    def self.from_share_error(message, code)
+      new_error = nil
+      KNOWN_ERRORS.each do |msg_str, error_class_name|
+        next unless body =~ /#{msg_str}/
```
Comment (pkarman) [inline lib/bgs/errors.rb]: where is `body` defined?

**Comment by pkarman** [comment]
@leikkisa should we move this forward or close?

**Comment by nanotone** [comment]
@pkarman I've resuscitated this PR by rebasing off of @jcq's latest work now in master, and slightly tidying up the known-error detection. Also, connecting to a more specific Github issue: department-of-veterans-affairs/caseflow#12166 -- this should be sufficient to ✔️ the first item in that issue.

I swapped around the `KNOWN_ERRORS` hash so the regex matching is more explicit, and subclass names are better hash keys anyway. The one regex improvement I had in mind is not actually a fancy one though: delete `none` from the POA error match, because sometimes we get a different value from BGS. Thoughts?

**Comment by leikkisa** [comment]
@nanotone I think we should keep `Power of Attorney of Folder is none`, because there may be other POA related errors that mean different things. Here is one with that text:
https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/5857/events/475987/

**Pull request merged**